packages/botbuilder-adapter-hangouts/src/request_validator.ts

```diff
@@ -0,0 +1,72 @@
+import fetch from 'cross-fetch';
+import { decode, verify } from 'jsonwebtoken';
+
+const issuer = 'chat@system.gserviceaccount.com';
+
+export class RequestValidator {
+    private url: string;
+    private expiresAt: Date;
+
+    private certificates: {[key: string]: string};
+
+    public constructor() {
+        this.url = `https://www.googleapis.com/service_accounts/v1/metadata/x509/${ issuer }`;
+        this.expiresAt = new Date();
+    }
+
+    /**
+     * Verifies that the token from the request infact was issued by Google.
+     * If we are unable to verify a token for any reason (malformed, expired, invalid) we
+     * return false.
+     * TODO: Replace this with an official implementation from googleapis if it becomes available.
```

Comment:
Ideally the google import will eventually provide this functionality but it doesn't currently appear to be available.